project-base/storefront/hooks/cart/useAddOrderItemsToCart.ts

```diff
@@ -0,0 +1,37 @@
+import { useCurrentCart } from 'connectors/cart/Cart';
+import { useAddOrderItemsToCartMutationApi } from 'graphql/generated';
+import { getInternationalizedStaticUrls } from 'helpers/localization/getInternationalizedStaticUrls';
+import { useDomainConfig } from 'hooks/useDomainConfig';
+import { useRouter } from 'next/router';
+import { Dispatch, SetStateAction, useState } from 'react';
+
+export const useAddOrderItemsToCart = (): [
+    string | undefined,
+    Dispatch<SetStateAction<string | undefined>>,
+    (orderUuid: string) => void,
+    (orderUuid: string, shouldMerge: boolean) => void,
+] => {
+    const [orderUuid, setOrderUuid] = useState<string>();
+    const { cart, isCartEmpty } = useCurrentCart();
+    const router = useRouter();
+    const { url } = useDomainConfig();
+    const [cartUrl] = getInternationalizedStaticUrls(['/cart'], url);
+    const [, AddOrderItemsToCart] = useAddOrderItemsToCartMutationApi();
+
+    const addOrderItemsToEmptyCart = (orderUuid: string) => {
+        if (isCartEmpty) {
+            AddOrderItemsToCart({ input: { orderUuid, cartUuid: null, shouldMerge: false } });
+            router.push(cartUrl);
+        } else {
+            setOrderUuid(orderUuid);
+        }
+    };
+
+    const mergeOrderItemsWithCurrentCart = (orderUuid: string, shouldMerge: boolean) => {
```

Comment:
`shouldMerge` doesn't seem to me as proper name since it feels to me like that it doesn't have to do something... :D What about `isMerge` or `isWithMerge`?

Reply:
It's also like that on BE. To me the name seems OK, as we use this naming convention in some other places, such as `shouldFocus`, `shouldShowMergeInfo`, `shouldBeShown`, etc. If you really dislike it, I can change it, but seems to me like an unnecessary change.

Reply:
I don't like it, but let's leave it then 🙃